src/components/routes/RoutesMapToolbar.vue

```diff
@@ -0,0 +1,209 @@
+ *
+ * @props
+ * - `drawEnabled` (boolean, required): Enabled state for draw tool.
+ * - `deleteEnabled` (boolean, required): Enabled state for delete tool.
+ *
+ * @events
+ * - `update:draw-enabled`: Emitted when draw tool is toggled.
+ * - `update:delete-enabled`: Emitted when delete tool is toggled.
+ * - `undo`: Emitted when undo button is clicked.
+ * - `save`: Emitted when save button is clicked.
+ * - `current-position`: Emitted when current position button is clicked.
+ *
+ * @example
+ * <routes-map-toolbar />
+ */
+
+// libraries
+import { defineComponent } from 'vue';
+
+export default defineComponent({
+  name: 'RoutesMapToolbar',
+  emits: [
+    'update:draw-enabled',
+    'update:delete-enabled',
+    'undo',
+    'save:route',
+    'current-position',
+  ],
+  props: {
+    drawEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    deleteEnabled: {
+      type: Boolean,
+      default: false,
+    },
+  },
+});
+</script>
+
+<template>
+  <div>
+    <!-- Toolbar: Top -->
+    <div
+      class="flex justify-center absolute-top q-pa-sm"
+      :style="{ zIndex: 1, pointerEvents: 'none' }"
+      data-cy="toolbar-top"
+    >
+      <q-toolbar
+        class="col-auto gap-8 q-pa-sm"
+        :style="{
+          borderRadius: '9999px',
+          backgroundColor: 'white',
+          pointerEvents: 'auto',
+        }"
+      >
+        <!-- Button: Enable draw (draw route) -->
+        <q-btn
+          dense
+          round
+          unelevated
+          class="q-pa-none q-ma-none"
+          color="transparent"
+          text-color="primary"
+          @click.prevent="$emit('update:draw-enabled', !drawEnabled)"
+          data-cy="add-route-button"
+        >
+          <q-avatar
+            size="32px"
+            class="q-pa-none q-ma-none"
+            :color="drawEnabled ? 'primary' : 'grey-3'"
+            data-cy="add-route-avatar"
+          >
+            <!-- Icon -->
+            <q-icon
+              name="mdi-pencil-plus"
+              :color="drawEnabled ? 'white' : 'primary'"
+              size="18px"
+              data-cy="add-route-icon"
+            />
+          </q-avatar>
+        </q-btn>
+        <!-- Button: Enable delete (delete point/vertex) -->
+        <q-btn
+          v-show="drawEnabled"
+          dense
+          round
+          unelevated
+          class="q-pa-none q-ma-none"
+          color="transparent"
+          text-color="primary"
+          @click.prevent="$emit('update:delete-enabled', !deleteEnabled)"
+          data-cy="delete-route-button"
+        >
+          <q-avatar
+            size="32px"
+            class="q-pa-none q-ma-none"
+            :color="deleteEnabled ? 'primary' : 'grey-3'"
+            data-cy="delete-route-avatar"
+          >
+            <!-- Icon -->
+            <q-icon
+              name="mdi-pencil-remove"
+              :color="deleteEnabled ? 'white' : 'primary'"
+              size="18px"
```

Comment:
Same as my comment above https://github.com/auto-mat/ride-to-work-by-bike-frontend/pull/470/files#r1682287232

Reply:
Updated.